python/demo/demo_mixed-poisson.py

```diff
@@ -109,88 +104,296 @@
+
+    opts = PETSc.Options()
+    opts[f"{ksp_sigma.prefix}pc_hypre_ams_cycle_type"] = 7
+    opts[f"{ksp_sigma.prefix}pc_hypre_ams_relax_times"] = 2
+
+    # Construct at set the 'discrete gradient' operator, which maps
+    # grad H1 -> H(curl), i.e. the gradient of a scalar Lagrange space
+    # to a H(curl) space
+    V_H1 = fem.functionspace(msh, element("Lagrange", msh.basix_cell(), k, dtype=xdtype))
+    V_curl = fem.functionspace(msh, element("N1curl", msh.basix_cell(), k, dtype=xdtype))
+    G = discrete_gradient(V_H1, V_curl)  #
+    G.assemble()
+    pc_sigma.setHYPREDiscreteGradient(G)
+
+    assert k > 0, "Element degree must be at least 1."
+    if k == 1:
+        # For the lowest order base (k=1), we can supply interpolation
+        # of the '1' vectors in the space V. Hypre can then construct
+        # the required operators from G and the '1' vectors.
+        cvec0, cvec1 = fem.Function(V), fem.Function(V)
+        cvec0.interpolate(lambda x: np.vstack((np.ones_like(x[0]), np.zeros_like(x[1]))))
+        cvec1.interpolate(lambda x: np.vstack((np.zeros_like(x[0]), np.ones_like(x[1]))))
+        pc_sigma.setHYPRESetEdgeConstantVectors(cvec0.x.petsc_vec, cvec1.x.petsc_vec, None)
     else:
-        raise e
+        # For high-order spaces, we must provide the (H1)^d -> H(div)
+        # interpolation operator/matrix
+        V_H1d = fem.functionspace(msh, ("Lagrange", k, (msh.geometry.dim,)))
+        Pi = interpolation_matrix(V_H1d, V)  # (H1)^d -> H(div)
+        Pi.assemble()
+        pc_sigma.setHYPRESetInterpolations(msh.geometry.dim, None, None, Pi, None)
+
+        # High-order elements generally converge less well than the
+        # lowest-order case with algebraic multigrid, so we perform
+        # extra work at the multigrid stage
+        opts[f"{ksp_sigma.prefix}pc_hypre_ams_tol"] = 1e-12
+        opts[f"{ksp_sigma.prefix}pc_hypre_ams_max_iter"] = 3
+
+    ksp_sigma.setFromOptions()
+else:
+    # If Hypre is not available, use LU factorisation on the $P_{00}$
+    # block
+    pc_sigma.setType("lu")
+# -
+
+# We create finite element functions that will hold the $\sigma$ and $u$
+# solutions:
+
+# +
+sigma, u = fem.Function(V, dtype=dtype), fem.Function(W, dtype=dtype)
+# -
+
+# Create a PETSc 'nested' vector that holds reference to the `sigma` and
+# `u` solution (degree-of-freedom) vectors and solve.
 
-sigma_h, u_h = w_h.split()
+# +
+x = PETSc.Vec().createNest([la.create_petsc_vector_wrap(sigma.x), la.create_petsc_vector_wrap(u.x)])
+ksp.solve(b, x)
+reason = ksp.getConvergedReason()
+assert reason > 0, f"Krylov solver has not converged {reason}."
+ksp.view()
+# -
+
+# We save the solution `u` in VTX format. DOLFINx VTX output support
+# visualisation of discontinuous Lagrange element, hence the finite
+# element field can be visualised exactly. However, VTX does not support
+# cell-wise data, so for the lowest-order `u` case we interpolate to a
+# degree 1 discontinuous space.
```

Comment:
```suggestion
This is not True any longer, ref: https://github.com/FEniCS/dolfinx/blob/fb0b99447b96434a48b44c4a256c29330ffac7c6/cpp/dolfinx/io/ADIOS2Writers.h#L587-L599
```
Paraview currently have some issues with visualizing DG-0 in time-dependent problems, but that is not on our implementation.